src/coreclr/jit/block.h

```diff
@@ -514,13 +514,20 @@ struct BasicBlock : private LIR::Range
 
     BBjumpKinds bbJumpKind; // jump (if any) at the end of this block
 
+    /* The following union describes the jump target(s) of this block */
+    union {
+        unsigned    bbJumpOffs; // PC offset (temporary only)
+        BasicBlock* bbJumpDest; // basic block
```

Comment:
You could imagine that `BBJ_COND` could have, here, `struct { BasicBlock* bbJumpTrue; BasicBlock* bbJumpFalse; }` where the true case would currently map to `bbJumpDest` and the false case to `bbNext`. And have accessors for each. Maybe that's where this will evolve?

Reply:
I think that's Andy's plan. He mentions in #93020 adding an explicit fall-through successor pointer to BasicBlock that would apply to BBJ_ALWAYS, BBJ_COND, etc. Would it be worth including that work in this PR, or save it for the next one? That change will probably introduce some TP diffs...

Reply:
Let's hold off... where I'd really like to get to is that these references are either:
* some kind of BasicBlockReference holder struct (so reassignment updates ref counts automagically); or even better, perhaps:
* FlowEdge(s)

The appeal of BasicBlockReference is that someday perhaps I'd like to be able to enumerate all the things that refer a block, including "external" BasicBlock references  as well (references from the EH table, from SSA defs, special block references from the compiler object (fgFirstBB, etc), etc). 

We could then eliminate all the fussing about with bbRefCounts and either have them maintained automatically or recomputed when needed.

The appeal of FlowEdges is that it simplifies some of the forward likelihood analysis, currently to find an edge to a successor you need to find the successor block and then enumerate the predecessor edges.

But not sure I want to do any of this just yet.

Reply:
While we are discussing successor edges, I'd like to bring up the fact there is duplication between IR control flow nodes such as `SWITCH`/`JTRUE` and the block kinds. In some systems, the information about successors is determined purely from the IR and "terminator" nodes.

It is an appealing system from the orthogonality/elegance perspective since there is no duplication, but it is unlikely to be suitable for us due to the throughput cost (of at least two additional indirections in HIR).